emcc.py

```diff
@@ -1441,10 +1436,18 @@ def has_c_source(args):
     if shared.Settings.DISABLE_EXCEPTION_THROWING and not shared.Settings.DISABLE_EXCEPTION_CATCHING:
       exit_with_error("DISABLE_EXCEPTION_THROWING was set (probably from -fno-exceptions) but is not compatible with enabling exception catching (DISABLE_EXCEPTION_CATCHING=0). If you don't want exceptions, set DISABLE_EXCEPTION_CATCHING to 1; if you do want exceptions, don't link with -fno-exceptions")
 
+    # Flags we pass to the compiler when building C/C++ code
+    # We add these to the user's flags (newargs), but not when building .s or .S assembly files
+    cflags = shared.get_cflags(newargs)
```

Comment:
wasn't this part of another change?

Reply:
Maybe you mean #10978 which just landed?   That change was about not passing link flags when compiling pch files.   This change is about not passing C flags (or link flags) when build .s files.

Reply:
Ah yes, it's not the same, you're right...

But like that one, it changes when `newargs` is assigned values. e.g. on line `1350 / 1345` newargs is appended to. So it seems like the order at least might change here?

Reply:
Yeah, those lines should probably be adding to cflags.  Let me do that too.

Reply:
OK, I converted all the ones that I could do.

This does mean that any cflags that we add internally here come before any user compile flags, which is think is what we want since then the user flags would take precendence.